stdlib/AllocProfile/src/AllocProfile.jl

```diff
@@ -0,0 +1,172 @@
+module AllocProfile
+
+using Base.StackTraces: StackTrace, StackFrame, lookup
+using Base: InterpreterIP
+
+# raw results
+
+# matches RawBacktrace on the C side
+struct RawBacktrace
+    data::Ptr{Csize_t} # in C: *jl_bt_element_t
+    size::Csize_t
+end
+
+# matches RawAlloc on the C side
+struct RawAlloc
+    type::Ptr{Type}
+    backtrace::RawBacktrace
+    size::Csize_t
+end
+
+struct TypeNamePair
+    addr::Csize_t
+    name::Ptr{UInt8}
+end
+
+struct FreeInfo
+    type::Ptr{Type}
+    count::UInt
+end
+
+# matches RawAllocResults on the C side
+struct RawAllocResults
+    allocs::Ptr{RawAlloc}
+    num_allocs::Csize_t
+
+    frees::Ptr{FreeInfo}
+    num_frees::Csize_t
+end
+
+function start(skip_every::Int=0)
+    ccall(:jl_start_alloc_profile, Cvoid, (Cint,), skip_every)
+end
+
+function stop()
+    raw_results = ccall(:jl_stop_alloc_profile, RawAllocResults, ())
+    decoded_results = GC.@preserve raw_results decode(raw_results)
+    ccall(:jl_free_alloc_profile, Cvoid, ())
+    return decoded_results
+end
+
+# decoded results
+
+struct Alloc
+    type::Type
+    stacktrace::StackTrace
+    size::Int
+end
+
+struct AllocResults
+    allocs::Vector{Alloc}
+    frees::Dict{Type,UInt}
+end
+
+const BacktraceEntry = Union{Ptr{Cvoid}, InterpreterIP}
+const BacktraceCache = Dict{BacktraceEntry,Vector{StackFrame}}
+
+# loading anything below this seems to segfault
+# TODO: find out what's going on
+TYPE_PTR_THRESHOLD = 0x0000000100000000
```

Comment:
Temporarily fixed the segfaults with this, hit for 2 out of ~65 types returned by my test workload (https://github.com/vilterp/HeapSnapshotParser.jl)